apache/superset
# fix: Add user filtering to changed_by. Fixes #27986

### SUMMARY
The EXTRA_RELATED_QUERY_FILTERS option only affected the "owner" field, this PR changes it to also apply to the "modified by" field.

This option is currently used in a superset installation where a superset user should not be able to list other superset user names. Without this PR a superset user can list the names of other users in the "modified by" field.

Fixes #27986

[Documentation](https://github.com/apache/superset/blob/4.0.2/superset/config.py#L1643-L1665) of the `EXTRA_RELATED_QUERY_FILTERS`:
```
# Extra related query filters make it possible to limit which objects are shown
# in the UI. For examples, to only show "admin" or users starting with the letter "b" in
# the "Owners" dropdowns, you could add the following in your config:
```

Example `superset_config.py` with EXTRA_RELATED_QUERY_FILTERS configured to hide all users:
```
# Hide all users
def user_filter(query):
    from sqlalchemy.sql import false
    return query.filter(false())

EXTRA_RELATED_QUERY_FILTERS = {
    "user": user_filter,
}
```

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:  
<img width="217" alt="image" src="https://github.com/apache/superset/assets/61765/1cfb06a8-faaf-4c81-b013-777092ef2721">

After:  
<img width="233" alt="image" src="https://github.com/apache/superset/assets/61765/b6ee4039-cbd0-46a4-89bd-0ed52ccee9e3">


### TESTING INSTRUCTIONS


1. Add the EXTRA_RELATED_QUERY_FILTERS above to your superset_config.py file
2. Click on "charts" to get the list of all charts
3. Click on the "owner" drop-down, only your name should be visible
4. Click on the "modified-by" drop-down, only your name should be visible (before this patch all users were visible) 

### ADDITIONAL INFORMATION


- [x] Has associated issue: #27986
